pkg/ambex/transforms.go

```diff
@@ -246,35 +246,50 @@ func V3ListenerToRdsListener(lnr *apiv3_listener.Listener) (*apiv3_listener.List
 // the supplied list. If there is no map entry for a given cluster, an empty ClusterLoadAssignment
 // will be synthesized. The result is a set of endpoints that are consistent (by the
 // go-control-plane's definition of consistent) with the input clusters.
-func JoinEdsClusters(ctx context.Context, clusters []ecp_cache_types.Resource, edsEndpoints map[string]*apiv2.ClusterLoadAssignment) (endpoints []ecp_cache_types.Resource) {
+func JoinEdsClusters(ctx context.Context, clusters []ecp_cache_types.Resource, edsEndpoints map[string]*apiv2.ClusterLoadAssignment, edsByPass bool) (endpoints []ecp_cache_types.Resource) {
 	for _, clu := range clusters {
 		c := clu.(*apiv2.Cluster)
 		// Don't mess with non EDS clusters.
 		if c.EdsClusterConfig == nil {
 			continue
 		}
 
-		// By default envoy will use the cluster name to lookup ClusterLoadAssignments unless the
+		// By default, envoy will use the cluster name to lookup ClusterLoadAssignments unless the
 		// ServiceName is supplied in the EdsClusterConfig.
 		ref := c.EdsClusterConfig.ServiceName
 		if ref == "" {
 			ref = c.Name
 		}
 
-		var source string
-		ep, ok := edsEndpoints[ref]
-		if ok {
-			source = "found"
+		// This change was introduced as a stop gap solution to mitigate the 503 issues when certificates are rotated.
+		// The issue is CDS gets updated and waits for EDS to send ClusterLoadAssignment.
+		// During this wait period calls that are coming through get hit with a 503 since the cluster is in a warming state.
+		// The solution is to "hijack" the cluster and insert all the endpoints instead of relying on EDS.
+		// Not there will be a discrepancy between envoy/envoy.json and the config envoy has
+		if edsByPass {
+			if ep, ok := edsEndpoints[ref]; ok {
+				c.LoadAssignment = ep
+				c.EdsClusterConfig = nil
+
+				// Type 1 is STRICT_DNS
+				c.ClusterDiscoveryType = &apiv2.Cluster_Type{Type: 1}
```

Comment:
Aren't the endpoints in this case already resolved to IP:Port so no DNS service discover is needed?

I would think we would just set it back to the default of 
 `STATIC`, no?
 https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#static?

Reply:
Good catch! 

I didn't think this one through. I changed it to Static (type 0) and confirmed it is working on a cluster.